nautobot/extras/tests/test_relationships.py

```diff
@@ -103,8 +105,38 @@ def test_clean_filter_not_valid(self):
             type=RelationshipTypeChoices.TYPE_MANY_TO_MANY,
         )
 
-        with self.assertRaises(ValidationError):
+        with self.assertRaises(ValidationError) as handler:
             m2m.clean()
+        expected_errors = {"source_filter": ["'notvalid' is not a valid filter parameter for dcim.Site object"]}
+        self.assertEqual(handler.exception.message_dict, expected_errors)
+
+        m2m = Relationship(
+            name="Another Vlan to Rack",
+            slug="vlan-rack-2",
+            source_type=self.site_ct,
+            source_filter={"site": "not a list"},
+            destination_type=self.rack_ct,
+            type=RelationshipTypeChoices.TYPE_MANY_TO_MANY,
+        )
+
+        with self.assertRaises(ValidationError) as handler:
+            m2m.clean()
+        expected_errors = {"source_filter": ["'site' is not a valid filter parameter for dcim.Site object"]}
+        self.assertEqual(handler.exception.message_dict, expected_errors)
+
+        m2m = Relationship(
+            name="Another Vlan to Rack",
+            slug="vlan-rack-2",
+            source_type=self.site_ct,
+            source_filter={"site": ["not a valid site"]},
+            destination_type=self.rack_ct,
+            type=RelationshipTypeChoices.TYPE_MANY_TO_MANY,
+        )
+
+        with self.assertRaises(ValidationError) as handler:
+            m2m.clean()
+        expected_errors = {"source_filter": ["'site' is not a valid filter parameter for dcim.Site object"]}
```

Comment:
Similarly, I would have expected the above test to complain because it's filtering on an invalid value, not an invalid key?